lib/oauth2.js

```diff
@@ -7,141 +7,109 @@ function Strategy(options, verify) {
   options = options || {};
   options.authorizationURL = options.authorizationURL || 'https://www.linkedin.com/oauth/v2/authorization';
   options.tokenURL = options.tokenURL || 'https://www.linkedin.com/oauth/v2/accessToken';
-  options.scope = options.scope || ['r_basicprofile'];
+  options.scope = options.scope || ['r_liteprofile'];
   options.profileFields = options.profileFields || null;
 
   //By default we want data in JSON
   options.customHeaders = options.customHeaders || {"x-li-format":"json"};
 
   OAuth2Strategy.call(this, options, verify);
+
+  this.options = options;
   this.name = 'linkedin';
-  this.profileUrl = 'https://api.linkedin.com/v1/people/~:(' + this._convertScopeToUserProfileFields(options.scope, options.profileFields) + ')';
+  this.profileUrl = 'https://api.linkedin.com/v2/me?projection=(id,firstName,lastName,profilePicture(displayImage~:playableStreams))';
+  this.emailUrl = 'https://api.linkedin.com/v2/emailAddress?q=members&projection=(elements*(handle~))';
 }
 
 util.inherits(Strategy, OAuth2Strategy);
 
 Strategy.prototype.userProfile = function(accessToken, done) {
-
   //LinkedIn uses a custom name for the access_token parameter
   this._oauth2.setAccessTokenName("oauth2_access_token");
 
   this._oauth2.get(this.profileUrl, accessToken, function (err, body, res) {
-    if (err) { return done(new InternalOAuthError('failed to fetch user profile', err)); }
-
     try {
+      if (err) {
+        throw new InternalOAuthError('failed to fetch user profile', err)
+      }
+
       var json = JSON.parse(body);
 
       var profile = { provider: 'linkedin' };
 
       profile.id = json.id;
-      profile.displayName = json.formattedName;
+
+      var firstName = json.firstName.localized[Object.keys(json.firstName.localized)[0]];
+      var lastName = json.lastName.localized[Object.keys(json.lastName.localized)[0]];
+
       profile.name = {
-                        familyName: json.lastName,
-                        givenName:  json.firstName
-                     };
-      profile.emails = [{ value: json.emailAddress }];
-      profile.photos = [];
-      if (json.pictureUrl) {
-          profile.photos.push({ value: json.pictureUrl });
+        givenName: firstName,
+        familyName: lastName
+      };
+
+      profile.displayName = firstName + ' ' + lastName;
+
+      if (
+        json.profilePicture &&
+        json.profilePicture['displayImage~'] &&
+        json.profilePicture['displayImage~'].elements &&
+        json.profilePicture['displayImage~'].elements.length > 0
+      ) {
+        profile.photos = json.profilePicture['displayImage~'].elements.reduce(function (memo, el) {
+          if (el && el.identifiers && el.identifiers.length > 0) {
+            memo.push({ value: el.identifiers[0].identifier }); // Keep the first pic for now
+          }
+          return memo;
+        }, []);
+      }
+
+      profile._profileRaw = body;
+      profile._profileJson = json;
```

Comment:
was wondering: why this breaking change?

Reply:
In order to conform with the Linkedin OAuth API, two separate requests need to be executed in order to retrieve the  user profile and then the email. 

Having code consistency in mind as well as taking into account the JS convention that says that keys staging with `_` are considered private, I kept both API raw and JSON responses under the following structure:

```js
profile._profileRaw = body;
profile._profileJson = json;
profile._emailRaw = body;
profile._emailJson = json;
```

Hope that helps.

Reply:
but passport already saved those under `profile._raw` and `profile._json`

As I'm using `profile._json` to get more data I had a bug when switching to your fork. That's how and why I'm calling it breaking. But it's fine as long as it is documented  :)